Spigot-Server-Patches/0625-Optimized-tick-ready-check.patch

```diff
@@ -0,0 +1,52 @@
++++ b/src/main/java/net/minecraft/server/World.java
+@@ -792,14 +792,13 @@ public abstract class World implements GeneratorAccess, AutoCloseable {
+
+             if (!tileentity.isRemoved() && tileentity.hasWorld()) {
+                 BlockPosition blockposition = tileentity.getPosition();
+-
+-                if (this.getChunkProvider().a(blockposition) && this.getWorldBorder().a(blockposition)) {
++                Chunk chunk; PlayerChunk playerChunk; if ((chunk = tileentity.getCurrentChunk()) != null && (playerChunk = chunk.playerChunk) != null && playerChunk.isTickingReady() && this.getWorldBorder().isInBounds(blockposition)) { // Paper - optimized tick ready check by inlining ChunkProviderServer.a(BlockPosition). Chunk lookup is no longer required and we can use the PlayerChunk directly available through the tile entity
+                     try {
+                         gameprofilerfiller.a(() -> {
+                             return String.valueOf(TileEntityTypes.a(tileentity.getTileType()));
+                         });
+                         tileentity.tickTimer.startTiming(); // Spigot
+-                        if (tileentity.getTileType().isValidBlock(this.getType(blockposition).getBlock())) {
++                        if (tileentity.getTileType().isValidBlock(chunk.getType(blockposition).getBlock())) { // Paper - reuse the chunk from above, do not look it up again
+                             ((ITickable) tileentity).tick();
+                         } else {
+                             tileentity.w();
+@@ -833,9 +832,11 @@ public abstract class World implements GeneratorAccess, AutoCloseable {
+                 this.tileEntityListTick.remove(tileTickPosition--);
+                 // Spigot end
+                 //this.tileEntityList.remove(tileentity); // Paper - remove unused list
+-                if (this.isLoaded(tileentity.getPosition())) {
+-                    this.getChunkAtWorldCoords(tileentity.getPosition()).removeTileEntity(tileentity.getPosition());
++                // Paper - prevent double chunk lookups
++                Chunk chunk; if ((chunk = this.getChunkIfLoaded(tileentity.getPosition())) != null) { // inlined contents of this.isLoaded(BlockPosition). Reuse the returned chunk instead of looking it up again
++                    chunk.removeTileEntity(tileentity.getPosition());
+                 }
++                // Paper end
+             }
+         }
+
+@@ -854,8 +855,8 @@ public abstract class World implements GeneratorAccess, AutoCloseable {
+                     }
+                     // CraftBukkit end */
+
+-                    if (this.isLoaded(tileentity1.getPosition())) {
+-                        Chunk chunk = this.getChunkAtWorldCoords(tileentity1.getPosition());
++                    Chunk chunk; if ((chunk = this.getChunkIfLoaded(tileentity1.getPosition())) != null) {  // inlined contents of this.isLoaded(BlockPosition). Reuse the returned chunk instead of looking it up again
```

Comment:
nitpick, these are two lines now, so two comments please :P

can fix that while merging tho